drivers: cc13xx_cc26xx: Convert drivers to use more new DT_INST macros

vanti opened the pull request:
Changing more of DT_* prefixed macros that refer to instances to use
DT_INST macros.

Verified with samples/subsys/console/echo and tests/drivers/spi/spi_loopback.

Signed-off-by: Vincent Wan <vincent.wan@linaro.org>

vanti committed:
drivers: cc13xx_cc26xx: Convert drivers to use more new DT_INST macros

Changing more of DT_* prefixed macros that refer to instances to use
DT_INST macros.

Signed-off-by: Vincent Wan <vincent.wan@linaro.org>

galak reviewed (changes requested):
So we need to figure out what to do with IOCPortConfigureSet().  I wonder for now if it would be better to either stick them in a board or soc specific file.  Than the drivers can truly be DT_INST based and when we have a proper pinmux/pinctrl API we can rework that.

vanti commented:
> So we need to figure out what to do with IOCPortConfigureSet(). I wonder for now if it would be better to either stick them in a board or soc specific file. Than the drivers can truly be DT_INST based and when we have a proper pinmux/pinctrl API we can rework that.

@galak Is the concern here that only macros prefixed with "DT_INST" should appear in the file, so I cannot use "DT_NODELABEL" directly without hiding it in another file? Just want to better understand the end goal before I rework this. IOCPortConfigureSet() is already a part of a "soc specific file" by definition since it is defined in the hal_ti repository.

vanti committed:
dts: bindings: cc13xx-cc26xx: add port info for uart tx and rx pins

Adding port information for uart tx and rx pins in dts, so that it can
be more systematically retrieved in the uart driver.

Signed-off-by: Vincent Wan <vincent.wan@linaro.org>

vanti committed:
drivers: cc13xx_cc26xx: use DT_INST_FOREACH to reduce code duplication

Use DT_INST_FOREACH macro to combine code used for multiple instances.
Remove unnecessary Kconfig options for UART instances.

Signed-off-by: Vincent Wan <vincent.wan@linaro.org>

vanti commented:
@galak Updated per our discussion to add port info to dts and use DT_INST_FOREACH.

A couple of things to note:

- I ended up defining GET_PIN/GET_PORT in the uart driver instead of in a common header such as soc/arm/ti_simplelink/cc13x2_cc26x2/soc.h, as the other drivers do not need port information for the APIs they are calling from TI driverlib.

- DEVICE_DEFINE seems to require the kernel level macros (e.g. PRE_KERNEL_1, POST_KERNEL) to be directly passed to it, so I ended up listing our every permutation. Ideas welcome if there is a way to further simplifying this.
